app/services/hyrax/multiple_membership_checker.rb

```diff
@@ -55,9 +55,12 @@ def single_membership_collections(collection_ids)
 
       field_pairs = {
         :id => collection_ids,
-        Hyrax.config.collection_type_index_field.to_sym => collection_type_gids_that_disallow_multiple_membership
+        Hyrax.config.collection_type_index_field.to_sym => collection_type_gids_that_disallow_multiple_membership&.map(&:to_s)
       }
-      Hyrax::FindObjectsViaSolrService.find_for_model_by_field_pairs(model: ::Collection, field_pairs: field_pairs, use_valkyrie: true)
+      Hyrax::SolrQueryService.new
+                             .with_model(model: ::Collection)
+                             .with_field_pairs(field_pairs: field_pairs, join_with: ' OR ')
+                             .get_objects(use_valkyrie: true).to_a
```

Comment:
The result is required to be an Array because later it is used in...

```
collections_to_check |= single_membership_collections(item.member_of_collection_ids)
```
where the result is `collections_to_check`.